catch/hipTestMain/main.cc

```diff
@@ -1,16 +1,50 @@
 #define CATCH_CONFIG_RUNNER
+#include <cmd_options.hh>
 #include <hip_test_common.hh>
 #include <iostream>
 
+CmdOptions cmd_options;
+
 int main(int argc, char** argv) {
   auto& context = TestContext::get(argc, argv);
   if (context.skipTest()) {
     // CTest uses this regex to figure out if the test has been skipped
     std::cout << "HIP_SKIP_THIS_TEST" << std::endl;
     return 0;
   }
-  int out = Catch::Session().run(argc, argv);
+
+  Catch::Session session;
+
+  using namespace Catch::clara;
+  // clang-format off
+  auto cli = session.cli() 
+    | Opt(cmd_options.iterations, "iterations")
+        ["-I"]["--iterations"]
+        ("Number of iterations used for performance tests (default: 1000)")
+    | Opt(cmd_options.warmups, "warmups")
+        ["-W"]["--warmups"]
+        ("Number of warmup iterations used for performance tests (default: 100)")
+    | Opt(cmd_options.no_display)
+        ["-S"]["--no-display"]
+        ("Do not display the output of performance tests")
+    | Opt(cmd_options.progress)
+        ["-P"]["--progress"]
+        ("Show progress bar when running performance tests")
+    | Opt(cmd_options.cg_extended_run, "cg_extened_run")
+        ["-E"]["--cg-extended-run"]
+        ("Number of iterations used for cooperative groups sync tests (default: 5)")
+    | Opt(cmd_options.accuracy_iterations, "accuracy_iterations")
+        ["-A"]["--accuracy-iterations"]
+        ("Number of iterations used for math accuracy tests with randomly generated inputs (default: 2^32)")
+    | Opt(cmd_options.accuracy_max_memory, "accuracy_max_memory")
+        ["-M"]["--accuracy-max-memory"]
+        ("Percentage of global device memory allowed for math accuracy tests (default: 80%)")
```

Comment:
How is this value going to be used?  Usually an absolute value is preferred since it's device independent.

Reply:
The runtime of the test can be sped up quite a bit when more memory is available. But since we don't want to hog all of the device memory, the --accuracy-max-memory option controls what percentage of the global memory can be used. We can make this an absolute value if that's what's preferred?

Reply:
Given how much variation there is, a percentage seems reasonable to me.